### DISCUSSION TITLE
Why does the BG of a Transparent image become black after cropping?

### DISCUSSION THREAD

**Discussion by osmanyz** [hm21/pro_image_editor#162] [Q&A] [answered]
I've been trying to figure out this but i'm not sure why it's happening.
I need your help in case of i'm missing any configuration.

This is a PNG image and I will need to allow users to crop their images but i faced this and it seems an issue but i'm not sure.

This screen record is captured here https://hm21.github.io/pro_image_editor/

**Comment by hm21** [comment, marked as answer]
The problem here is that the default output format is `JPEG`. You can change it to `PNG` as below, which should resolve the problem.

```dart
ProImageEditor.memory(
        bytes!,
        configs: const ProImageEditorConfigs(
          imageGenerationConfigs: ImageGeneratioConfigs(
            outputFormat: OutputFormat.png,
          ),
        ),
),
```

**Reply by osmanyz** [reply]
Thank you, it worked. I'm just curious why JPG is a default type. Shouldn't the image type be the default type?

**Reply by hm21** [reply]
There are a few reasons, but the main reason is performance and platform support.

The `JPEG` format is faster compared to `PNG` in the generation progress in dart and the encoder is easy to understand, which made it easy for me to edit the encoder a bit so that I can destroy an active encoding progress even if it's not finished. This is very useful in case the user edits too fast and overloads the background processors with too many tasks.

The editor also didn't support modern formats like WebP and AVIF, which also need to fall back to another format, which might confuse the user when the editor sometimes returns the correct format and sometimes not depending on the image format.

I also think in most cases the user will upload the image to the server where a background function will convert it into multiple formats for the best platform support or maybe use an image loader API like ImageKit, so the local format just needs to be fast in generation with good quality.

However, for the case you want that it autodetect your format, I recommend you to use the package [mime](https://pub.dev/packages/mime) which allows you to detect the image format. You can then choose, depending on the mime type, which output format you want to use. I also have this [example](https://github.com/hm21/pro_image_editor/blob/155d1fe4ca084c4bf4ac7c303febd32d99d008a4/example/lib/pages/image_format_convert_example.dart#L36-L44) which shows you how you can convert the image format to WebP even though the editor didn't support it as an output format.

**Reply by osmanyz** [reply]
Hello @hm21 I'm glad that you've spend your time to explain it. I sincerely appreciate your help. Thank you so much.
